src/Microsoft.DotNet.Interactive/CompositeKernel.cs

```diff
@@ -213,6 +213,22 @@ protected override void SetHandlingKernel(KernelCommand command, KernelInvocatio
                 throw new NoSuitableKernelException(command);
             }
 
+            switch (command)
```

Comment:
Rather than duplicating all of this, why not move it to the base and put the actual cancellation logic into a virtual method?